scripts/update_schemastore.py

```diff
@@ -7,49 +7,78 @@
 
 from __future__ import annotations
 
+import enum
 import json
 from pathlib import Path
 from subprocess import check_call, check_output
 from tempfile import TemporaryDirectory
+from typing import NamedTuple
 
-schemastore_fork = "git@github.com:astral-sh/schemastore.git"
-schemastore_upstream = "git@github.com:SchemaStore/schemastore.git"
 ruff_repo = "https://github.com/astral-sh/ruff"
 root = Path(
     check_output(["git", "rev-parse", "--show-toplevel"], text=True).strip(),
 )
 ruff_json = Path("schemas/json/ruff.json")
 
 
-def update_schemastore(schemastore: Path) -> None:
-    if not schemastore.is_dir():
-        check_call(["git", "clone", schemastore_fork, schemastore])
+class SchemastoreRepos(NamedTuple):
+    fork: str
+    upstream: str
+
+
+class GitProtocol(str, enum.Enum):
+    SSH = "ssh"
+    HTTPS = "https"
+
+    def schemastore_repos(self) -> SchemastoreRepos:
+        match self:
+            case GitProtocol.SSH:
+                return SchemastoreRepos(
+                    fork="git@github.com:astral-sh/schemastore.git",
+                    upstream="git@github.com:SchemaStore/schemastore.git",
+                )
+            case GitProtocol.HTTPS:
+                return SchemastoreRepos(
+                    fork="https://github.com/astral-sh/schemastore.git",
+                    upstream="https://github.com/SchemaStore/schemastore.git",
+                )
+            case _:
+                assert False, "Should be unreachable"
```

Comment:
Pyright is unhappy with this branch locally because it's unreachable (which is kinda the point). It stops complaining if I use `typing.assert_never` instead of `assert False`, but that's only available on Python 3.11+. Are we okay with requiring Python 3.11 to run this script? Using pattern-matching already means that the script requires Python 3.10, I suppose

Reply:
Could `typing_extensions` be used instead?

Reply:
it definitely could, but I don't really wanna add a third-party dependency for a one-file script that we run as part of the release process -- it would mean you'd either need to activate a venv and have the requirements installed into it, or run the script using `uv run`, etc. Doesn't seem worth it for something as small as this :-)

Reply:
If we do care about keeping compatibility with Python 3.10, it would actually be pretty simple to do our own backport that would satisfy pyright, though: https://github.com/python/cpython/blob/main/Tools/cases_generator/_typing_backports.py. I could do that, but it seems like more boilerplate, and I'm not sure if we care about keeping compatibility with Python 3.10 😆

Reply:
Feel free to require Python 3.11+ for the script

Reply:
Imho our internal scripts should be latest stable only, we have it installed for dev work anyway and it's good when we get to practice the new features ourselves

Reply:
I updated it to use `assert_never`, but pyright still complains locally. I think this is a false positive; mypy specifically excludes branches that are obviously _meant_ to be unreachable from its reachability checks. I left a comment at https://github.com/microsoft/pyright/issues/7559#issuecomment-2108198425 about this.